python/cudf/cudf/tests/test_text.py

```diff
@@ -829,3 +831,15 @@ def test_read_text_byte_range_large(datadir):
         f.write(content)
 
     cudf.read_text(temp_file, delimiter=delimiter)
+
+
+def test_read_text_in_memory(datadir):
+    delimiter = "::"
+
+    # Since Python split removes the delimiter and read_text does
+    # not we need to add it back to the 'content'
+    expected = cudf.Series(["x::", "y::", "z"])
+
+    actual = cudf.read_text(StringIO("x::y::z"), delimiter=delimiter)
```

Comment:
I think you could define the delimiter inline.
```suggestion
    # Since Python split removes the delimiter and read_text does
    # not we need to add it back to the 'content'
    expected = cudf.Series(["x::", "y::", "z"])

    actual = cudf.read_text(StringIO("x::y::z"), delimiter="::")
```